addons/sale/models/sale.py

```diff
@@ -113,6 +113,20 @@ def _compute_tax_id(self):
             order.order_line._compute_tax_id()
 
     def _search_invoice_ids(self, operator, value):
+        if operator == 'in' and value:
+            self.env.cr.execute("""
+                SELECT array_agg(so.id)
+                    FROM sale_order so
+                    JOIN sale_order_line sol ON sol.order_id = so.id
+                    JOIN sale_order_line_invoice_rel soli_rel ON soli_rel.order_line_id = sol.id
+                    JOIN account_move_line aml ON aml.id = soli_rel.invoice_line_id
+                    JOIN account_move am ON am.id = aml.move_id
+                WHERE
+                    am.type in ('out_invoice', 'out_refund') AND
+                    am.id = ANY(%s)
+            """, (list(value),))
+            so_ids = self.env.cr.fetchone()[0] or []
```

Comment:
Is the `or []` not useless there?

Reply:
Since we are using a `array_agg` to aggregate all the results into a single row, we have to pay attention about the `array_agg` behavior.
The postgresql [documentation says this about the aggregate functions](https://www.postgresql.org/docs/9.5/functions-aggregate.html)

> It should be noted that except for count, these functions return a null value when no rows are selected. In particular, sum of no rows returns null, not zero as one might expect, and array_agg returns null rather than an empty array when there are no input rows. The coalesce function can be used to substitute zero or an empty array for null when necessary.

The code would crash if at the end the domain we return is `[('id', 'in', False)]` if no sale orders is found :/